components/kubelet/pod/component.go

```diff
@@ -108,36 +116,51 @@ func (c *component) CheckOnce() {
 		c.lastMu.Unlock()
 	}()
 
-	// assume "kubelet" is not installed, thus not needed to check its activeness
 	if c.checkDependencyInstalled == nil || !c.checkDependencyInstalled() {
+		// "kubelet" is not installed, thus not needed to check its activeness
 		return
 	}
 
-	// below are the checks in case "kubelet" is installed, thus requires activeness checks
-	cctx, ccancel := context.WithTimeout(c.ctx, 15*time.Second)
-	running := checkKubeletReadOnlyPortListening(cctx, c.kubeletReadOnlyPort)
-	ccancel()
-	if !running {
-		d.err = errors.New("kubelet is installed but kubelet is not running")
+	if c.checkKubeletRunning == nil || !c.checkKubeletRunning() {
+		// "kubelet" is not running, thus not needed to check its activeness
 		return
 	}
 
-	if c.checkServiceActive != nil {
-		var err error
-		cctx, ccancel = context.WithTimeout(c.ctx, 15*time.Second)
-		d.KubeletServiceActive, err = c.checkServiceActive(cctx)
-		ccancel()
-		if !d.KubeletServiceActive || err != nil {
-			d.err = fmt.Errorf("kubelet is installed but kubelet service is not active or failed to check (error %v)", err)
-			return
-		}
-	}
-
-	cctx, ccancel = context.WithTimeout(c.ctx, 30*time.Second)
+	// below are the checks in case "kubelet" is installed and running, thus requires activeness checks
+	cctx, ccancel := context.WithTimeout(c.ctx, 30*time.Second)
 	d.NodeName, d.Pods, d.err = listPodsFromKubeletReadOnlyPort(cctx, c.kubeletReadOnlyPort)
 	ccancel()
 
-	d.connErr = isConnectionRefusedError(d.err)
+	if d.err != nil {
+		c.failedCount++
+	} else {
+		c.failedCount = 0
+	}
+
+	c.lastHealthy = d.err == nil
+	if isConnectionRefusedError(d.err) && c.ignoreConnectionErrors {
```

Comment:
I'm wondering which case should set `c.ignoreConnectionErrors=false`..  we cannot ignore connection errors since we have checked if the kubelet opens readonly port..

Reply:
The idea was, we don't want to alert when the kubelet restarts, but now that we check port open, think we can safely assume we don't need this?

/cc @cardyok

Reply:
There's a scene we might miss as we rely on checking port..

Reply:
I thought about it carefully, let's do check if the kubelet bin is there..

Reply:
addressed